terraform/context_test.go

```diff
@@ -697,43 +697,6 @@ func TestContext2Plan_preventDestroy_destroyPlan(t *testing.T) {
 	}
 }
 
-func TestContext2Plan_providerAliasMissing(t *testing.T) {
```

Comment:
This whole test was testing the opposite of the behavior I added. It doesn't make a lot of sense to me that this should be an error, because it breaks modules completely. I think we just default it like every other provider type.